identity-iota/src/credential/presentation_validator.rs

```diff
@@ -0,0 +1,857 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::BTreeMap;
+
+use identity_core::common::OneOrMany;
+use identity_credential::credential::Credential;
+use identity_credential::presentation::Presentation;
+use identity_did::verifiable::VerifierOptions;
+use serde::Serialize;
+
+use super::errors::AccumulatedCredentialValidationError;
+use super::errors::AccumulatedPresentationValidationError;
+use super::errors::ValidationError;
+use super::PresentationValidationOptions;
+use crate::credential::credential_validator::CredentialValidator;
+use crate::did::IotaDID;
+use crate::document::ResolvedIotaDocument;
+use crate::Error;
+use crate::Result;
+
+/// A struct for validating [Presentation]s.
+#[non_exhaustive]
+pub struct PresentationValidator;
```

Comment:
Most of the comments for `CredentialValidator` apply here too, like the function naming scheme, potentially unnecessary  `_local_error` functions etc.